### PR TITLE
feat(review-workflows): add default workflow

### PR DESCRIPTION
**Pull request by nathan-pichon**
### What does it do?

Adding a default workflow on bootstrap automatically if there isn't any.
The default stages have been decided by the Content Squad and are the following: To do, Ready to review, In progress, and Reviewed.

This workflow will be able to be edited in further developments.

### Why is it needed?

We want users only to have one workflow for the review workflow feature.
This is the only workflow that will be available in the MVP.

### How to test it?

You can get this workflow through Strapi commands querying the `admin::workflow` model.

### Related issue(s)/PR(s)

#15594 needs to be merged before.

### CODE REVIEW

File: packages/core/admin/ee/server/constants/default-stages.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[
+  {
+    "name": "To do"
+  },
+  {
+    "name": "In progress"
+  },
+  {
+    "name": "Done"
+  }
```
[the remark this answers is not in the document]
Comment (nathan-pichon) [reply packages/core/admin/ee/server/constants/default-stages.json:10]: Sorry, I don't know why I had `To do, In progress, Done` 🤔 
Good catch, thanks.

File: packages/core/admin/ee/server/services/review-workflows/review-workflows.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  async bootstrap() {
+    const wfCount = await strapi.entityService.count(WORKFLOW_MODEL_UID);
+    const stagesCount = await strapi.entityService.count(STAGE_MODEL_UID);
+
+    if (wfCount + stagesCount === 0) {
+      const defaultStages = require('../../constants/default-stages.json');
+      const defaultWorkflow = require('../../constants/default-workflow.json');
+
+      await strapi.query('admin::workflow-stage').createMany({ data: defaultStages });
+
```
Comment (Marc-Roig) [inline packages/core/admin/ee/server/services/review-workflows/review-workflows.js:15]: This being the service layer,  it makes sense for it to use the DB layer.

Still, I personally use we should use the entityService whenever we can (when performance is not critial), as it has the stablest interface.

wdyt?

File: packages/core/admin/ee/server/services/review-workflows/review-workflows.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const { WORKFLOW_MODEL_UID, STAGE_MODEL_UID } = require('../../constants/workflows');
+
+module.exports = ({ strapi }) => ({
+  async bootstrap() {
+    const wfCount = await strapi.entityService.count(WORKFLOW_MODEL_UID);
+    const stagesCount = await strapi.entityService.count(STAGE_MODEL_UID);
+
+    if (wfCount + stagesCount === 0) {
```
Comment (Marc-Roig) [inline packages/core/admin/ee/server/services/review-workflows/review-workflows.js:10]: why looking at the stages count? can we have 0 workflows and more than 0 stages?

File: packages/core/admin/ee/server/services/review-workflows/review-workflows.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const { WORKFLOW_MODEL_UID, STAGE_MODEL_UID } = require('../../constants/workflows');
+
+module.exports = ({ strapi }) => ({
+  async bootstrap() {
```
Comment (Marc-Roig) [inline packages/core/admin/ee/server/services/review-workflows/review-workflows.js:6]: I think this is the best way to define the bootstrap for RW right now.
But I also think the EE folder is not structured to be scalable. We could discuss it with more people, but I think we could find a way to define the bootstrap outside a service.

What I miss in the EE folder is separating features in different folders like:
```
.
├── audit-logs/
│   ├── controllers
│   ├── routes
│   └── ...
├── sso/
│   ├── controllers
│   ├── routes
│   └── ...
└── review-workflows/
    ├── controllers
    ├── routes
    └── ...
```

File: packages/core/admin/ee/server/services/review-workflows/review-workflows.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+    if (wfCount + stagesCount === 0) {
+      const defaultStages = require('../../constants/default-stages.json');
+      const defaultWorkflow = require('../../constants/default-workflow.json');
+
+      await strapi.query('admin::workflow-stage').createMany({ data: defaultStages });
+
+      const stages = await strapi
+        .query('admin::workflow-stage')
+        .findMany({ limit: 4, select: ['id'] });
```
Comment (Convly) [inline packages/core/admin/ee/server/services/review-workflows/review-workflows.js:18]: Is it a choice to make this static and not dynamic (talking about the 4 & the relations connect operations).
Can't we do smth like `limit: defaultStages.length` and connect: /* smth dynamic based on the id/idx */ ?

File: packages/core/admin/ee/server/services/review-workflows/review-workflows.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (nathan-pichon) [reply packages/core/admin/ee/server/services/review-workflows/review-workflows.js:10]: For MVP, if there is anything in the DB, the feature is already bootstrap and we shouldn't do anything.

If there are already some stages or workflows in the DB, it means the user has manually created those and we shouldn't create a new workflow "starter default pack".

Comment (nathan-pichon) [reply packages/core/admin/ee/server/services/review-workflows/review-workflows.js:10]: If there are any stages or workflows, we already have our default workflow, or the user has tricked his database.

File: packages/core/admin/ee/server/services/review-workflows/review-workflows.js (hunk 4 above)
Comment (nathan-pichon) [reply packages/core/admin/ee/server/services/review-workflows/review-workflows.js:6]: I agree, we could have a more modular folder approach. But I don't know if we want to organize our code like that. Should it be a collegial decision? Like, BE sync?

Comment (Marc-Roig) [reply packages/core/admin/ee/server/services/review-workflows/review-workflows.js:6]: agree, this should be in the BE sync. Let's bring the topic next friday :)

File: packages/core/admin/ee/server/services/review-workflows/review-workflows.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  return {
+    async bootstrap() {
+      const wfCount = await workflowsService.count();
+      const stagesCount = await stagesService.count();
+
+      // Check if there is nothing about review-workflow in DB
+      // If any, the feature has already been initialized with a workflow and stages
+      if (wfCount === 0 && stagesCount === 0) {
+        const defaultStages = require('../../constants/default-stages.json');
+        const defaultWorkflow = require('../../constants/default-workflow.json');
```
Comment (Marc-Roig) [inline packages/core/admin/ee/server/services/review-workflows/review-workflows.js:40]: In the future, will we allow users to define the default stages / workflows ? 

Just a thought for the future, we could let users define this constants in the plugin configuration. wdyt?

**Review by Marc-Roig — APPROVED** [review]
LGTM :)

File: packages/core/admin/ee/server/services/review-workflows/review-workflows.js (hunk 6 above)
Comment (nathan-pichon) [reply packages/core/admin/ee/server/services/review-workflows/review-workflows.js:40]: This is not a plugin, but surely, we can think of a way to configure the default workflow of RW feature. 👍

**Pull request merged**